src/escape.rs

```diff
@@ -0,0 +1,45 @@
+use std::borrow::{Borrow, Cow};
+
+enum Escaped<'a> {
+    Mut(&'a str),
+    Id,
+}
+
+fn escape_char<'a>(c: char) -> Escaped<'a> {
+    match c {
+        '<'  => Escaped::Mut("&lt;"),
+        '>'  => Escaped::Mut("&gt;"),
+        '"'  => Escaped::Mut("&quot;"),
+        '\'' => Escaped::Mut("&apos;"),
+        '&'  => Escaped::Mut("&amp;"),
+        _    => Escaped::Id,
+    }
+}
+
+pub fn escape_text<'a>(s: &Cow<'a, str>) -> Cow<'a, str> {
```

Comment:
This may not be the best design, but I chose this because I thought repeated single-character writes (like in maud) may incur a performance penalty.

Another promising design came to my mind just now :smile: ...